File: C++/121. Best Time to Buy and Sell Stock.cpp
#include <vector>

using namespace std;

int maxProfit(vector<int> &prices)
{
    int buy = prices[0], max_profit = 0;

    for (auto price : prices)
    {
        max_profit = max(max_profit, price - buy);
        buy = min(buy, price);
    }

    return max_profit;
}

File: Java/121. Best Time to Buy and Sell Stock.java
class Solution {

    public int maxProfit(int[] prices) {
        int profit = 0;
        int bought = Integer.MAX_VALUE;

        for (var price : prices) {
            bought = Integer.min(price, bought);
            profit = Integer.max(price - bought, profit);
        }

        return profit;
    }
}